### ISSUE TITLE
"Cannot set hostname SSL binding for {domain}"-error after successful execution

### ISSUE THREAD

**Issue by ThomasBleijendaal** [wearetriple/ados-extension-ssl-custom-domain#11]
The step returns 

![image](https://user-images.githubusercontent.com/132195/150093506-2daff70d-7e3f-43f7-9a23-ab57f3b486cd.png)

while adding the custom domain and ssl certificate in Azure successfully. This only happens for new environments that have their custom domain and ssl configured for the first time.

The next time the step is run the execution is successful and there are no errors.

**Comment by rdeveen** [comment]
I think this also happens when you need to update the certificate because a renewal.

**Comment by ThomasBleijendaal** [comment]
Ok I found some extra details why this happens to my pipeline.

The step cannot assign a PFX to a function app which has its service plan in a different resource group. If a PFX is added to a function app of that same service plan that resides in the same resource group first, then this step can assign the PFX in 2 attempts. 

The SSL binding step probably uses the wrong resource group as it perhaps needs to use the resource group of the service plan, instead of the function app. It's all a bit weird as PFX resources in Azure are shared (between resource groups, I don't know?) and live as separate resources outside function apps / web apps, but are only managed inside a function app / web app.

**Comment by ThomasBleijendaal** [comment]
@rdeveen Can you elaborate the issue you encounter because of the renewal? Is de PFX not updated correctly?

**Comment by rdeveen** [comment]
Yes I will, if it happens again. I should open a new issue then.
